vhost_rs/src/vhost_user/master.rs

```diff
@@ -361,7 +361,8 @@ impl VhostUserMaster for Master {
         // But what content should the payload contains for a get_config() request?
         // So current implementation doesn't conform to the spec.
         let hdr = node.send_request_with_config_body(MasterReq::GET_CONFIG, &body, None)?;
-        let (reply, buf, rfds) = node.recv_reply_with_payload::<VhostUserConfig>(&hdr)?;
+        let (reply, buf, rfds) =
+            node.recv_reply_with_payload::<VhostUserConfig>(body.size as usize, &hdr)?;
```

Comment:
Even if this works, this is not solving the root of the issue. Please take a look at my branch https://github.com/sboeuf/cloud-hypervisor-1/tree/fix-vu-blk
I took your branch, I reverted two of your patches, and I fixed the `GET_CONFIG` command, which now means you shouldn't run into the issue you were seeing.

Reply:
I tried your branch, it is still failed in recvmsg for get_config info from backend.

as for the revert patch: **https://github.com/sboeuf/cloud-hypervisor-1/commit/e74e0cae02f500139b225795f4e35b5d521633c2**,  i also had another solution in vhost crate after my vhost_rs changes in cloud-hypervisor.https://github.com/jiangliu/vhost/commit/d5a42d1ea580dd46dcd6274bc67d2a5f970488c9

Reply:
Yes I can see that this patch you're mentioning is very close to my patch. The difference is that I expect the caller to provide the array of bytes (the payload) instead of providing an empty array from the vhost_rs crate.
The only thing I realized I'm missing in my branch is this line https://github.com/jiangliu/vhost/commit/d5a42d1ea580dd46dcd6274bc67d2a5f970488c9#diff-70449da7a94b8102ba670b5029bf011bR559
Could you try by fixing this line?

Also, is the failure you're seeing the same you were seeing before?

Reply:
Could you please try with my updated branch. I have added this line https://github.com/sboeuf/cloud-hypervisor-1/commit/927f16f017efa2b39c1f13f768c62a357b3b5166#diff-739b90af7d2cfe98f8eb767f56a0aad2R555 to my patch, making the buffer being the right size, which should prevent `readmsg()` from being stuck.